### PR TITLE
WIP: bump(*): update etcd to 3.3.3

### PR DESCRIPTION
**Pull request by mfojtik**
This bumps the etcd in master branch to 3.3.3 which should match what our ansible will be installing soon. 

(also pins go4.org to use their github repo as their website is down?)

### CODE REVIEW

File: glide.yaml [hunk 1]
```diff
@@ -181,6 +181,13 @@ import:
 - package: google.golang.org/appengine
   version: 12d5545dc1cfa6047a286d5e853841b6471f4c19
 
+  # to avoid go4.org website outages
+- package: go4.org
```
Comment (mfojtik) [inline glide.yaml]: this mitigates the issues from having go4.org site down... see: https://github.com/kubernetes/kubernetes/issues/39868

File: vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/storage/etcd/testing/utils.go [hunk 2]
```diff
@@ -188,7 +188,7 @@ func configureTestCluster(t *testing.T, name string, https bool) *EtcdTestServer
 // launch will attempt to start the etcd server
 func (m *EtcdTestServer) launch(t *testing.T) error {
 	var err error
-	if m.s, err = etcdserver.NewServer(&m.ServerConfig); err != nil {
+	if m.s, err = etcdserver.NewServer(m.ServerConfig); err != nil {
```
Comment (liggitt) [inline vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/storage/etcd/testing/utils.go]: this makes me sad... we *just* got to the point where openshift/kuberenetes could be built/run

File: glide.lock [hunk 3]
```diff
@@ -385,14 +385,16 @@ imports:
...
+  version: bb3d318650d48840a39aa21a027c6630e198e626
 - name: github.com/elazarl/go-bindata-assetfs
   version: 3dcc96556217539f50599357fb481ac0dc7439b9
 - name: github.com/emicklei/go-restful
   version: ff4f55a206334ef123e4f79bbf348980da81ca46
   subpackages:
   - log
 - name: github.com/emicklei/go-restful-swagger12
-  version: 5e28dc7157ebd9247a268c26dd1bee7818aa33f7
+  version: 14b2c9246e6851e0297450a3d5c7288c0485592e
```
Comment (liggitt) [inline glide.lock]: where is this SHA from? this looks like it is moving us off our branch: https://github.com/openshift/emicklei-go-restful-swagger12/commits/openshift-3.9

File: vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/storage/etcd/testing/utils.go (hunk 2 above)
Comment (mfojtik) [reply vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/storage/etcd/testing/utils.go]: @liggitt do you know how far is upstream with bumping the etcd?

File: glide.lock (hunk 3 above)
Comment (mfojtik) [reply glide.lock]: @liggitt hrm weird, I thought this is transitive etcd dependency. Have we pinned this in glide?

File: glide.lock [hunk 4]
```diff
@@ -1053,14 +1055,16 @@ imports:
 - name: github.com/sirupsen/logrus
   version: 89742aefa4b206dcf400792f3bd35b542998eb3b
 - name: github.com/skynetservices/skydns
-  version: 775ef406696380ce92b76e72421e86b6868905d8
+  version: 04111322ced82c1520abbb1053745272f715bd5b
```
[the remark this answers is not in the document]
Comment (mfojtik) [reply glide.lock]: Will dig what is going on there

File: vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/storage/etcd/testing/utils.go (hunk 2 above)
Comment (liggitt) [reply vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/storage/etcd/testing/utils.go]: looks like plan is for 3.3 in 1.11: https://github.com/kubernetes/kubernetes/issues/61326#issuecomment-374737257

Comment (mfojtik) [reply vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/storage/etcd/testing/utils.go]: @ingvagabund @liggitt in that case, I would prefer to wait for 3.11 with bumping the etcd client, so we can get upstream to update the deps and the config structs, which will mean we won't risk nasty regressions and we will also be able to build the 1.10 openshift/kubernetes.

File: glide.lock (hunk 4 above)
[the remark this answers is not in the document]
Comment (mfojtik) [reply glide.lock]: @liggitt ok, so here is weird... I ran the ./hack/update-deps.sh in my Linux VM and everything seems to work normally. I don't see any changes in go-restful or skydns... When I run this on Mac directly, with the same exact version of glide, I'm getting different results, even when I clean vendor/ directory...

I was also not able to trace that SHA anywhere, neither my local GIT repos have that or Github or Kube or Origin so I have no idea where the glide is getting this SHA from. 

Another weirdness I saw just on OSX is: https://gist.github.com/mfojtik/bce873326d7badf7ab302d73f9840f29

On Linux, I haven't seen this conflict.

**Comment by mfojtik** [comment]
Closing this as we are not going to bump etcd to 3.3.3 in 3.10 cycle but will do this when kube switch to 3.3 as well.

**Pull request closed without merging**